### PR TITLE
FLOE-584: Update to latest Flocking and Infusion

### PR DESCRIPTION
**Pull request by colinbdclark**
This PR updates the Chart Authoring tool to the latest versions of its dependencies. We also rename the module and version number to be npm complaint so that we can publish it to NPM.

This pull request is based on @waharnum's #19, but does not yet address the code review comments by @cindyli. We are happy to address these at some point, but will need to talk a bit with @waharnum first.

Pair programmed by me and @ptcher as part of the process of getting the Nexus demos back up and running.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,31 @@
-##What Is the Chart Authoring Tool?##
+## What Is the Chart Authoring Tool?##
```
Comment (jobara) [inline README.md:1]: You can remove the "##" from the end of the heading.

File: README.md [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
 
 A tool for authoring accessible charts for the web. Including a sonified (audio) version of the chart.
 
-##How Do I Get the Chart Authoring Tool?##
+## How Do I Get the Chart Authoring Tool?##
```
Comment (jobara) [inline README.md]: Same as above, you can remove the "##" from the end.

File: README.md (hunk 1 above)
Comment (jobara) [reply README.md:1]: Same with the other headings that use this style.

File: demos/index.html [hunk 3]
```diff
@@ -7,6 +7,9 @@
         <link rel="stylesheet" type="text/css" href="../src/lib/infusion/src/framework/core/css/fluid.css" />
         <link rel="stylesheet" type="text/css" href="../src/lib/infusion/src/components/overviewPanel/css/OverviewPanel.css" />
         <link rel="stylesheet" type="text/css" href="../src/lib/infusion/src/components/inlineEdit/css/InlineEdit.css" />
+        <link rel="stylesheet" type="text/css" href="../../node_modules/infusion/src/lib/jquery/ui/css/default-theme/core.css" />
+        <link rel="stylesheet" type="text/css" href="../../node_modules/infusion/src/lib/jquery/ui/css/default-theme/tooltip.css" />
+        <link rel="stylesheet" type="text/css" href="../../node_modules/infusion/src/lib/jquery/ui/css/default-theme/theme.css" />
```
Comment (jobara) [inline demos/index.html:12]: These are returning 404 errors.

File: demos/src/js/chartAuthoring-demo-overview.js [hunk 4]
```diff
@@ -41,7 +57,10 @@ var demo = demo || {};
             storeModel: {
                 type: "fluid.modelComponent",
                 options: {
-                    gradeNames: ["fluid.prefs.cookieStore"],
+                    gradeNames: [
+                        "fluid.prefs.cookieStore",
+                        "fluid.prefs.cookieStore.writable"
+                    ],
```
Comment (jobara) [inline demos/src/js/chartAuthoring-demo-overview.js:63]: There doesn't seem to be anything stored in the cookie. I tried toggling the overview panel closed, changed values for the chart, and set the title and description. None of those are saved in the cookie and when a page is reloaded they all return to their default states.

Comment (jobara) [reply demos/src/js/chartAuthoring-demo-overview.js:63]: From looking at the code, it does look like it should only be storing the state of the OverViewPanel. I think the problem is actually with the model setup for this component. I've left a comment in the related section.

File: demos/src/js/chartAuthoring-demo-overview.js [hunk 5]
```diff
@@ -41,7 +57,10 @@ var demo = demo || {};
...
                     model: {
                         expander: {
                             funcName: "{that}.get"
```
Comment (jobara) [inline demos/src/js/chartAuthoring-demo-overview.js:66]: The cookie store is now a dataSource. DataSource [get](https://github.com/fluid-project/infusion/blob/a79da1b9c996e6546896d468e8f30ef7aa1325ea/src/framework/core/js/DataSource.js#L177-L182) and [set](https://github.com/fluid-project/infusion/blob/a79da1b9c996e6546896d468e8f30ef7aa1325ea/src/framework/core/js/DataSource.js#L194-L207) methods return promises. These promises are resolved with the values. This expander is expecting the stored data to be returned.

**Comment by jobara** [comment]
Is there an easy way to test the various chat types and chart transitions? I don't see a demo for those.

**Comment by jobara** [comment]
Regarding the following comment in the PR description:

> This pull request is based on @waharnum's #19, but does not yet address the code review comments by @cindyli. We are happy to address these at some point, but will need to talk a bit with @waharnum first.

Do you intend to address @cindyli's comments in this PR after you talk to @waharnum or do you mean in subsequent work in a new PR later?

**Comment by jobara** [comment]
> Pair programmed by me and @ptcher as part of the process of getting the Nexus demos back up and running.

You can setup your git commit to include @ptcher as a [co-author](https://help.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors).

File: demos/src/js/chartAuthoring-demo-overview.js [hunk 6]
```diff
@@ -19,6 +19,22 @@ var demo = demo || {};
...
+    //    in order to configure which buttons appear within it
+    // 2. There is a "magic file" for including the OverviewPanel
+    //    in demos that are part of Infusion, which we can't access
+    //    since it is pruned out of the Infusion npm module.
+    //
+    // In addition, for some reason, the Chart Authoring Tool adds
+    // an additional (probably quite useful?) feature where the state
+    // of the OverviewPanel is persisted in a cookie, so that if the
+    // user dismisses it, the panel won't reappear the next time they
+    // visit the demo. Perhaps this is a feature that could be
```
Comment (amb26) [inline demos/src/js/chartAuthoring-demo-overview.js:36]: Was always intended

File: src/js/lineChartTimeSeriesArea.js [hunk 7]
```diff
@@ -132,7 +140,22 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
                 "transform": "translate(0)"
             });
         });
+    };
 
+    // Uses D3's default transition; this results in the
+    // "wriggle" described at https://bost.ocks.org/mike/path/
+    floe.chartAuthoring.lineChart.timeSeries.area.updateArea.defaultTransition = function (chartLineAreaPaths, area, width, transitionLength) {
```
Comment (amb26) [inline src/js/lineChartTimeSeriesArea.js:147]: Seems identical to the one in "Line" series - can be consolidated?

**Pull request closed without merging**